app/controllers/oauth2_controller.rb

```diff
@@ -5,19 +5,25 @@
 # @!visibility private
 class Oauth2Controller < ApplicationController
 
+  PASSWORD_PROVIDER = 'password'.freeze
+  FACEBOOK_PROVIDER = 'facebook'.freeze
+  GOOGLE_PROVIDER = 'google'.freeze
+  EDX_PROVIDER = 'edx'.freeze
+  DEFAULT_CLIENT = 'unspecified'.freeze
+
   force_ssl if: -> { RailsApiAuth.force_ssl }
 
   # rubocop:disable MethodLength
   def create
     case params[:grant_type]
     when 'password'
-      authenticate_with_credentials(params[:username], params[:password])
+      authenticate_with_credentials(params[:username], params[:password], PASSWORD_PROVIDER, params[:accessing_application])
     when 'facebook_auth_code'
-      authenticate_with_facebook(params[:auth_code])
+      authenticate_with_facebook(params[:auth_code], FACEBOOK_PROVIDER, params[:accessing_application])
```

Comment:
Do we really need to send this as a parameter if the function has facebook in the name? I would think you could just use it in the authenticate_with_* methods